lib/model/app_model/decode_invoice.dart

```diff
@@ -0,0 +1,48 @@
+import 'package:cln_common/cln_common.dart';
+import 'package:clnapp/model/app_model/app_utils.dart';
+
+class AppDecodeInvoice {
+  DecodeInvoice invoice;
+  AppDecodeInvoice(this.invoice);
+
+  factory AppDecodeInvoice.fromJSON(Map<String, dynamic> json,
+      {bool snackCase = false}) {
+    LogManager.getInstance.debug("RESPONSE HERE : $json");
+    var decodedInvoice = DecodeInvoice.fromJSON(json);
+    return AppDecodeInvoice(decodedInvoice);
+  }
+}
+
+class DecodeInvoice {
+  String description;
+
+  String expirationTime;
+
+  String pubKey;
+
+  int amount;
+
+  String createdTime;
+
+  DecodeInvoice(
+      {required this.description,
+      required this.amount,
+      required this.expirationTime,
+      required this.pubKey,
+      required this.createdTime});
+
+  factory DecodeInvoice.fromJSON(Map<String, dynamic> json,
+      {bool snackCase = false}) {
+    var description = json.withKey("description", snackCase: snackCase);
+    var expirationTime = json.withKey("expiry", snackCase: snackCase);
+    var amount = json.withKey("amount_msat", snackCase: true);
+    var pubKey = json.withKey("payee", snackCase: snackCase);
+    var createdTime = json.withKey("created_at", snackCase: true);
```

Comment:
please, keep attention to the `snakeCase`

Reply:
There are two things to it :

Either we can do something like `var amount = json.withKey("amount_msat", snackCase: !snackCase);`
or something like : `var amount = json.withKey("amountMsat", snackCase: snackCase);`

Reply:
Why? What is it happening here?

The snackCase is true for the GRPC, and false for the other kind of client, if you are receive true for the other client you are introducing a bug somewhere

Reply:
Actually the string we passed to the `withKey` method will try to convert the `createdAt` to `created_at` when the `snackCase` is false to retrieve the data from the json we get from the `unix socket`.

Reply:
in the unix socket `snackCase` should be true